lib/puppet/provider/postgresql_psql/ruby.rb

```diff
@@ -16,21 +16,19 @@ def run_sql_command(sql)
     command.push("-p", resource[:port]) if resource[:port]
     command.push("-t", "-c", '"' + sql.gsub('"', '\"') + '"')
 
-    environment = get_environment
-
     if resource[:cwd]
       Dir.chdir resource[:cwd] do
-        run_command(command, resource[:psql_user], resource[:psql_group], environment)
+        run_command(command, resource[:psql_user], resource[:psql_group])
       end
     else
-      run_command(command, resource[:psql_user], resource[:psql_group], environment)
+      run_command(command, resource[:psql_user], resource[:psql_group])
     end
   end
 
   private
 
   def get_environment
-    environment = resource[:connect_settings] || {}
+    environment = resource[:connect_settings].dup || {}
```

Comment:
This doesn't work when `resource[:connect_settings]` is empty. I guess `environment = environment.dup` on the next line is better, but I'll verify that when I have a test environment set up.